### PR TITLE
Upgraded specs2 to 5.0.0-RC-07 for Scala3

### PR DESCRIPTION
**Pull request by etorreborre**
and reformatted the fragments a bit to look better when using Scala 3.

This PR prepares `discipline-specs2` for the upcoming `specs2-5.0.0` release.

### CODE REVIEW

**Comment by vasilmkd** [comment]
This turned out to be easier than expected?

**Comment by etorreborre** [comment]
Well, unless I missed something :-)

**Comment by vasilmkd** [comment]
This looks good to me, but I will wait for another review from @djspiewak and @armanbilge.

**Comment by vasilmkd** [comment]
The final thing to consider is whether to release stable versions that depend on release candidates.

**Comment by etorreborre** [comment]
I propose to wait a bit until I publish the final `specs2-5.0.0`. I need to address the [`cats-effect-testing` library first](https://github.com/typelevel/cats-effect-testing/issues/190).

File: build.sbt [hunk 1]
```diff
@@ -113,10 +114,7 @@ lazy val commonSettings = Seq(
   libraryDependencies += "org.typelevel" %%% "discipline-core" % disciplineV,
   libraryDependencies += {
     if (isDotty.value)
-      ("org.specs2" %%% "specs2-scalacheck" % specs2V)
-        .cross(CrossVersion.for3Use2_13)
-        .exclude("org.scalacheck", "scalacheck_2.13")
-        .exclude("org.scalacheck", "scalacheck_sjs1_2.13")
+      "org.specs2" %%% "specs2-scalacheck" % specs2DottyV
     else
       "org.specs2" %%% "specs2-scalacheck" % specs2V
```
Comment (djspiewak) [inline build.sbt:119]: This makes me really nervous because it makes it very hard for users to easily cross-build between Scala 2 and Scala 3.

**Comment by armanbilge** [comment]
> The final thing to consider is whether to release stable versions that depend on release candidates.

@vasilmkd would you mind publishing a snapshot so I can test this downstream? :)

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:119]: @rossabaker suggested in https://github.com/typelevel/discipline-specs2/issues/192#issuecomment-907707311 that the next version of discipline-specs2 is Scala 3 only.

Comment (djspiewak) [reply build.sbt:119]: That's probably reasonable. It'll be more of a problem for cats-effect-testing though

**Comment by vasilmkd** [comment]
@armanbilge 

Snapshot release will be available soon.

```scala
"io.vasilev" %%% "discipline-specs2" % "1.2-7-e3ce260"
```

**Comment by armanbilge** [comment]
Awesome, thanks! Trying now.

**Comment by armanbilge** [comment]
Cool, works in https://github.com/armanbilge/cheshire/runs/3546471023. Trying another one of my projects next.

**Comment by armanbilge** [comment]
Update, also working in https://github.com/armanbilge/schrodinger/runs/3547259110. That's more interesting maybe because it tests monad transformers against CE3 laws.

@etorreborre I just realized the discipline output is sometimes missing newlines, e.g. https://github.com/armanbilge/cheshire/runs/3546471023#step:6:70

**Comment by etorreborre** [comment]
@armanbilge thanks I thought I had tested that but I must have missed the mutable spec case. I will fix it.

**Comment by etorreborre** [comment]
Actually I'm not sure if there's a meaningful fix in `discipline-specs2`. The easiest thing, if you are using a mutable spec is to intersperse some `br`s in between the `checkAll` calls:
```
class MutableDisciplineSpec extends Specification with Discipline {
  checkAll("Int1", RingLaws.ring); nl
  checkAll("Int2", RingLaws.ring); nl
  checkAll("Int3", RingLaws.ring)

  def nl = { br; br }
}
```
I tried to write a `display` function doing this interspersing but given the amount of side effects in a mutable spec this is close to impossible!

The simplest thing is actually to use a non-mutable spec:
```
class DisciplineSpec extends Specification with Discipline {
  def is =
      checkAll("Int1", RingLaws.ring) ^
      checkAll("Int2", RingLaws.ring) ^
      checkAll("Int3", RingLaws.ring)
}
```

**Comment by armanbilge** [comment]
@etorreborre thanks for looking into this so quickly. Apologies if I'm missing something, but can't the `nl` just be baked into the `checkAll` method?

**Comment by etorreborre** [comment]
If you do it then you have an extra `nl` at the end of the spec even if there's only `checkAll` being called. If that's ok with you, you can define:
```
def check(name, prop) = { checkAll(name, prop); br; br }
```
Having some sort of wysiwig display in specs2 is quite hard, either when side-effects are involved with a mutable spec or with a `s2""" ... """` string in an immutable spec (to respect newlines and whitespace). I tried to improve the situation in specs2-5.x but there are definitely some corner cases that will pop-up.

**Comment by armanbilge** [comment]
> If you do it then you have an extra `nl` at the end of the spec even if there's only `checkAll` being called.

IMO we should make that change in this PR. An extra `nl` is preferable to I suspect nearly every downstream having to account for this one way or another. Thoughts @djspiewak @vasilmkd?

**Comment by etorreborre** [comment]
I prefer to be able to control exactly how things are displayed but the user is king :-).

**Pull request closed without merging**